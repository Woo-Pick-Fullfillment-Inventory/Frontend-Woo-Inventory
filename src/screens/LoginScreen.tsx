import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { signin, logout, SigninPayload } from '../redux/authSlice';
import { Text, TextInput, Button } from 'react-native-paper';
import { RootState } from '../redux/store';

const LoginScreen = () => {
  const dispatch = useDispatch();
  const isLoggedIn = useSelector((state: RootState) => state.auth.isLoggedIn);

  // State hooks for email and password
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleLogin = () => {
    // const userData: SigninPayload = {
    //   /* data from form later */
    // };
    // dispatch(signin(userData));
  };

  const handleLogout = () => {
    dispatch(logout());
  };

  return (
    <>
      <TextInput
        label="Email"
        value={email}
        onChangeText={text => setEmail(text)}
        autoCapitalize="none"
      />
      <TextInput
        label="Password"
        value={password}
        onChangeText={text => setPassword(text)}
        secureTextEntry
        autoCapitalize="none"
      />
      <Button onPress={handleLogin}>Login</Button>
      {isLoggedIn && <Button onPress={handleLogout}>Logout</Button>}

      <Text>This is the login page</Text>
    </>
  );
};

export default LoginScreen;
