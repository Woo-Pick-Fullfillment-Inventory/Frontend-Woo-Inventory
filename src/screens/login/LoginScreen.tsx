import { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { signin, logout, SigninPayload } from '../../redux/authSlice';
import { StyleSheet, View, Text, GestureResponderEvent } from 'react-native';

import { InputField } from '../../components/InputField';
import { Button } from '../../components/Button';
import { CheckBox } from '../../components/CheckBox';
import { RootState } from '../../redux/store';
import {globalStyle} from '../../theme'
import Logo from '../../static/logo/wi.svg'

import { login } from './utils';

export const LoginScreen = () => {
  const [checked, setChecked] = useState(false);
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  return (
    <View style={globalStyle.screenContainer}>
      <Logo width={100} />

      <View style={globalStyle.section}>
        <Text style={globalStyle.heading1} >Login</Text>

        <View style={styles.intro}>
          <Text style={globalStyle.paragraph}>The ultimate solution for</Text>
          <Text style={globalStyle.paragraph}>optimal warehouse management</Text>
        </View>
      </View>

      <View style={globalStyle.section}>
        <InputField
          title={'Email'}
          placeholder={'Input your email address'}
          value={email}
          action={(text) => setEmail(text)}
        />
        <InputField
          title={'Password'}
          placeholder={'Password'}
          value={password}
          action={(text) => setPassword(text)}
          isPassword
        />
       
        <Text>
          <CheckBox
            isChecked={checked}
            title={'Remember me'}
            action={() => setChecked(!checked)}
          />
          </Text>
      </View>
     
      <View style={styles.buttonSection}>
        <Button 
          onPress={() => login(email, password)} 
          title={'Login'}
        />
      </View>
    </View>
  );
};

export type Props = {
  title: string,
  placeholder: string,
};

const styles = StyleSheet.create({
  intro: {
    marginTop: 10
  },

  buttonSection: {
    marginTop: 15,
  },

  checkbox: {
    width: 2
  }
});

export default LoginScreen;

