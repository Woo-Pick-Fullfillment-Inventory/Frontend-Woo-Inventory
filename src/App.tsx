import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import ProfileScreen from './screens/ProfileScreen';
import MainMenuScreen from './screens/MainMenuScreen';
import WelcomeScreen from './screens/WelcomeScreen';
import LoginScreen from './screens/LoginScreen';
import SignupScreen from './screens/SignupScreen';
import ScannerScreen from './screens/ScannerScreen';
import { BLACKCOLOR } from './theme';

export type RootStackParamList = {
  HomeScreen: undefined; // No parameters expected
  AuthScreen: undefined;
  LoginScreen: undefined;
  SignupScreen: undefined;
  ProfileScreen: { name: string };
  MainMenuScreen: undefined;
  WelcomeScreen: undefined;
  AgbScreen: undefined;
  ScannerScreen: undefined;
};

const Stack = createNativeStackNavigator<RootStackParamList>();

const App = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          name="WelcomeScreen"
          component={WelcomeScreen}
          options={{ title: '', headerShown: false }}
        />
        <Stack.Screen
          name="LoginScreen"
          component={LoginScreen}
          options={{
            title: '',
            headerShadowVisible: false,
            headerTintColor: BLACKCOLOR,
          }}
        />
        <Stack.Screen
          name="SignupScreen"
          component={SignupScreen}
          options={{
            title: '',
            headerShadowVisible: false,
            headerTintColor: BLACKCOLOR,
          }}
        />
        <Stack.Screen
          name="ProfileScreen"
          component={ProfileScreen}
          options={{ title: 'Profile' }}
        />
        <Stack.Screen
          name="MainMenuScreen"
          component={MainMenuScreen}
          options={{
            title: 'MainMenuScreen',
            gestureEnabled: false, // So that user can not swipe back in IOS
            headerBackVisible: false
          }}
        />
        <Stack.Screen
          name="ScannerScreen"
          component={ScannerScreen}
          options={{ title: 'Scanner Screen' }}
        />
        <Stack.Screen name="AgbScreen" component={ScannerScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default App;
