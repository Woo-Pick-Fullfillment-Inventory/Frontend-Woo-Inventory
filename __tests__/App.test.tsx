import * as React from 'react';
import { create } from 'react-test-renderer';
import App from '../src/App';
import LoginScreen from 'src/screens/Login';

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: jest.fn(),
    // Mock other navigation methods your component uses
  }),
}));
jest.mock('@react-navigation/native-stack', () => ({
  createNativeStackNavigator: jest.fn(),
}));

describe('Testing App', () => {
  it('renders correctly', () => {
    const tree = create(<App />);
    expect(tree).toMatchSnapshot();
  });
});

test('renders correctlyyyyyyyyyyy', () => {
  const tree = create(<LoginScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
