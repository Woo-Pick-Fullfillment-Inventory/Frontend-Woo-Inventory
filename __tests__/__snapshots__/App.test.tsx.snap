// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  />
</View>
`;
